jobapi/routes.go

```diff
@@ -0,0 +1,133 @@
+		HeadersMdlw(),
+		AuthMdlw(s.token),
+	)
+
+	r.Route("/api/current-job/v0", func(r chi.Router) {
+		r.Get("/env", s.getEnv)
+		r.Patch("/env", s.patchEnv)
+		r.Delete("/env", s.deleteEnv)
+	})
+
+	return r
+}
+
+func (s *Server) getEnv(w http.ResponseWriter, _ *http.Request) {
+	normalizedEnv := s.environ.Dump()
+	resp := EnvGetResponse{Env: normalizedEnv}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(resp)
+}
+
+func (s *Server) patchEnv(w http.ResponseWriter, r *http.Request) {
+	var req EnvUpdateRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	defer r.Body.Close()
+	if err != nil {
+		err = fmt.Errorf("failed to decode request body: %w", err)
+		writeError(w, err, http.StatusBadRequest)
+		return
+	}
+
+	added := make([]string, len(req.Env))
+	updated := make([]string, len(req.Env))
+	protected := checkProtected(maps.Keys(req.Env))
+
+	if len(protected) > 0 {
+		err = fmt.Errorf("the following environment variables are protected, and cannot be modified: % v", protected)
+		writeError(w, err, http.StatusUnprocessableEntity)
+		return
+	}
+
+	nils := make([]string, 0, len(req.Env))
+
+	for k, v := range req.Env {
+		if v == nil {
+			nils = append(nils, k)
+		}
+	}
+
+	if len(nils) > 0 {
+		err = fmt.Errorf("removing environment variables (ie setting them to null) is not permitted on this endpiont. The following keys were set to null: % v", nils)
+		writeError(w, err, http.StatusUnprocessableEntity)
+		return
+	}
+
+	for k, v := range req.Env {
+		if _, ok := s.environ.Get(k); ok {
+			updated = append(updated, k)
+		} else {
+			added = append(added, k)
+		}
+		s.environ.Set(k, *v)
+	}
+
+	resp := EnvUpdateResponse{
+		Added:   added,
+		Updated: updated,
+	}
+
+	json.NewEncoder(w).Encode(resp)
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *Server) deleteEnv(w http.ResponseWriter, r *http.Request) {
+	var req EnvDeleteRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	defer r.Body.Close()
+	if err != nil {
+		err = fmt.Errorf("failed to decode request body: %w", err)
+		writeError(w, err, http.StatusBadRequest)
+		return
+	}
+
+	protected := checkProtected(req.Keys)
+	if len(protected) > 0 {
+		err = fmt.Errorf("the following environment variables are protected, and cannot be modified: % v", protected)
+		writeError(w, err, http.StatusUnprocessableEntity)
+		return
+	}
+
+	deleted := make([]string, 0, len(req.Keys))
+	for _, k := range req.Keys {
+		if _, ok := s.environ.Get(k); ok {
+			deleted = append(deleted, k)
+			delete(s.environ, k)
+		}
+	}
+
+	json.NewEncoder(w).Encode(EnvDeleteResponse{Deleted: deleted})
+	w.WriteHeader(http.StatusOK)
+}
+
+func checkProtected(candidates []string) []string {
+	protected := make([]string, 0, len(candidates))
+	for _, c := range candidates {
+		if _, ok := agent.ProtectedEnv[c]; ok {
+			protected = append(protected, c)
+		}
+	}
+	return protected
+}
+
+func writeError(w http.ResponseWriter, err error, code int) {
```

Comment:
I noticed that 4 out of 9 times, `writeError` is called with a string wrapped in an error - I'm guessing solely to satisfy the type of `err` here? Since this is more or less a convenience function, I reckon it makes sense to be a bit looser with the arg type. Use `any` and stringify it with `fmt.Sprint`?